From 44428ad14fda11c07dae69dc69f5f18d8172cae8 Mon Sep 17 00:00:00 2001
From: gaohuatao <gaohuatao@huawei.com>
Date: Thu, 15 Oct 2020 09:23:19 +0800
Subject: [PATCH 01/28] iSulad: modify defattr to 755 in spec

Signed-off-by: gaohuatao <gaohuatao@huawei.com>
---
 iSulad.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iSulad.spec b/iSulad.spec
index 35bd125..534d616 100644
--- a/iSulad.spec
+++ b/iSulad.spec
@@ -198,7 +198,7 @@ fi
 %{_includedir}/isulad/*
 %attr(0755,root,root) %{_libdir}/pkgconfig
 %attr(0640,root,root) %{_libdir}/pkgconfig/isulad.pc
-%defattr(0550,root,root,0750)
+%defattr(0755,root,root,0755)
 %{_bindir}/*
 %{_libdir}/*
 %attr(0640,root,root) %{_sysconfdir}/sysconfig/iSulad
-- 
2.20.1

