From 9e2556703a44d96feb66972541ce200cd018801a Mon Sep 17 00:00:00 2001
From: jikui <jikui2@huawei.com>
Date: Tue, 23 Jun 2020 20:59:09 +0800
Subject: [PATCH 2/6] iSulad: resolve coredump of isula inspect ""

Signed-off-by: jikui <jikui2@huawei.com>
---
 src/cmd/isula/information/inspect.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/cmd/isula/information/inspect.c b/src/cmd/isula/information/inspect.c
index e9b6ff6d..fe011ad9 100644
--- a/src/cmd/isula/information/inspect.c
+++ b/src/cmd/isula/information/inspect.c
@@ -199,8 +199,9 @@ static int client_inspect_container(const struct isula_inspect_request *request,
 
     ret = ops->container.inspect(request, response, config);
     if (ret != 0) {
-        if ((strstr(response->errmsg, "Inspect invalid name") != NULL) ||
-            (strstr(response->errmsg, "No such image or container or accelerator") != NULL)) {
+        if ((response->errmsg != NULL) &&
+            (strstr(response->errmsg, "Inspect invalid name") != NULL ||
+            strstr(response->errmsg, "No such image or container or accelerator") != NULL)) {
             return CONTAINER_NOT_FOUND;
         }
 
@@ -814,6 +815,7 @@ int cmd_inspect_main(int argc, const char **argv)
     free(filter_string);
 
     if (status) {
+        COMMAND_ERROR("Inspec error: No such object:%s", g_cmd_inspect_args.name);
         exit(ECOMMON);
     }
     exit(EXIT_SUCCESS);
-- 
2.25.1

