From e0c13b23ddee161a6196ca2e0c970704e4bbef0e Mon Sep 17 00:00:00 2001
From: WangFengTu <wangfengtu@huawei.com>
Date: Sun, 24 Apr 2022 14:18:26 +0800
Subject: [PATCH 3/3] support isula wait even if it's not oci image

Signed-off-by: WangFengTu <wangfengtu@huawei.com>
---
 src/cmd/isula/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/isula/main.c b/src/cmd/isula/main.c
index 0e5b845c..2deb862c 100644
--- a/src/cmd/isula/main.c
+++ b/src/cmd/isula/main.c
@@ -134,11 +134,11 @@ struct command g_commands[] = {
         "rmi", false, cmd_rmi_main, g_cmd_rmi_desc, NULL, &g_cmd_rmi_args
     },
 #endif
-#ifdef ENABLE_OCI_IMAGE
     {
         // `wait` sub-command
         "wait", false, cmd_wait_main, g_cmd_wait_desc, NULL, &g_cmd_wait_args
     },
+#ifdef ENABLE_OCI_IMAGE
     {
         // `logs` sub-command
         "logs", false, cmd_logs_main, g_cmd_logs_desc, NULL, &g_cmd_logs_args
-- 
2.25.1

