From 0ddc58b78b0ca7d6c1cb52b10e3fa03f0da69326 Mon Sep 17 00:00:00 2001
From: zhangxiaoyu <zhangxiaoyu58@huawei.com>
Date: Sat, 15 Oct 2022 16:51:38 +0800
Subject: [PATCH 05/39] add extern C for mainloop header

Signed-off-by: zhangxiaoyu <zhangxiaoyu58@huawei.com>
---
 src/utils/cutils/mainloop.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/utils/cutils/mainloop.h b/src/utils/cutils/mainloop.h
index 5124b33a..7a4f1cfd 100644
--- a/src/utils/cutils/mainloop.h
+++ b/src/utils/cutils/mainloop.h
@@ -18,6 +18,10 @@
 #include <stdint.h>
 #include "linked_list.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef void (*epoll_timeout_callback_t)(void *data);
 
 struct epoll_descr {
@@ -42,4 +46,8 @@ extern int epoll_loop_open(struct epoll_descr *descr);
 
 extern int epoll_loop_close(struct epoll_descr *descr);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
-- 
2.25.1

