From abffe70870a2867d423dcf24f4b97791b1df16bb Mon Sep 17 00:00:00 2001
From: lifeng68 <lifeng68@huawei.com>
Date: Thu, 22 Oct 2020 11:25:35 +0800
Subject: [PATCH 06/28] info: fix typo driverr to driver

Signed-off-by: lifeng68 <lifeng68@huawei.com>
---
 src/cmd/isula/information/info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/isula/information/info.c b/src/cmd/isula/information/info.c
index 07cad9d..d6f6f7b 100644
--- a/src/cmd/isula/information/info.c
+++ b/src/cmd/isula/information/info.c
@@ -76,7 +76,7 @@ static void client_info_server(const struct isula_info_response *response)
         printf("Logging Driver: %s\n", response->logging_driver);
     }
     if (response->cgroup_driver != NULL) {
-        printf("Cgroup Driverr: %s\n", response->cgroup_driver);
+        printf("Cgroup Driver: %s\n", response->cgroup_driver);
     }
     if (response->huge_page_size != NULL) {
         printf("Hugetlb Pagesize: %s\n", response->huge_page_size);
-- 
2.20.1

