From 9133e2159e9e69434b41e9649762f8eaed191f37 Mon Sep 17 00:00:00 2001
From: zhangxiaoyu <zhangxiaoyu58@huawei.com>
Date: Tue, 31 May 2022 22:39:44 +0800
Subject: [PATCH 09/28] remove static of strlncat

Signed-off-by: zhangxiaoyu <zhangxiaoyu58@huawei.com>
---
 src/utils/http/parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/http/parser.c b/src/utils/http/parser.c
index 0e0e603b..12df2435 100644
--- a/src/utils/http/parser.c
+++ b/src/utils/http/parser.c
@@ -47,7 +47,7 @@
 #include "utils.h"
 #include "isula_libutils/log.h"
 
-static size_t strlncat(char *dststr, size_t size, const char *srcstr, size_t nsize)
+size_t strlncat(char *dststr, size_t size, const char *srcstr, size_t nsize)
 {
     size_t ssize, dsize;
 
-- 
2.25.1

