From 05fee32dd6faaa154d7a8a2e44b6b6153515f223 Mon Sep 17 00:00:00 2001
From: wujing <Jing.Woo@outlook.com>
Date: Tue, 15 Sep 2020 16:43:18 +0800
Subject: [PATCH 10/10] fix: bad formatting placeholder in http parse module

Signed-off-by: wujing <Jing.Woo@outlook.com>
---
 src/utils/http/parser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/http/parser.c b/src/utils/http/parser.c
index ec4ae43..2a69fed 100644
--- a/src/utils/http/parser.c
+++ b/src/utils/http/parser.c
@@ -303,7 +303,7 @@ int parse_http(const char *buf, size_t len, struct parsed_http_message *m,
 
     nparsed = parse(buf, len, parser);
     if (nparsed != len) {
-        ERROR("Failed to parse it, parsed :%ld, intput:%ld \n", nparsed, len);
+        ERROR("Failed to parse it, parsed :%zu, intput:%zu \n", nparsed, len);
         ret = -1;
         goto free_out;
     }
-- 
2.25.1

