From 861a635c95254c7429bc8c23552f529c5a083762 Mon Sep 17 00:00:00 2001
From: WangFengTu <wangfengtu@huawei.com>
Date: Thu, 2 Dec 2021 15:24:31 +0800
Subject: [PATCH 14/14] fix undefined reference in libisulad_img.so

Signed-off-by: WangFengTu <wangfengtu@huawei.com>
---
 src/daemon/modules/image/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/daemon/modules/image/CMakeLists.txt b/src/daemon/modules/image/CMakeLists.txt
index 86b7d8a6..0a004835 100644
--- a/src/daemon/modules/image/CMakeLists.txt
+++ b/src/daemon/modules/image/CMakeLists.txt
@@ -71,6 +71,7 @@ add_library(${LIB_ISULAD_IMG} ${LIBTYPE}
     ${CMAKE_SOURCE_DIR}/src/utils/tar/isulad_tar.c
     ${CMAKE_SOURCE_DIR}/src/utils/tar/util_archive.c
     ${CMAKE_SOURCE_DIR}/src/utils/tar/util_gzip.c
+    ${CMAKE_SOURCE_DIR}/src/daemon/config/isulad_config.c
     )
 
 target_include_directories(${LIB_ISULAD_IMG} PUBLIC
-- 
2.25.1

