From de8ef6a226fdbee53975d6d746a065a24a98ea05 Mon Sep 17 00:00:00 2001
From: WangFengTu <wangfengtu@huawei.com>
Date: Mon, 6 Dec 2021 11:07:36 +0800
Subject: [PATCH] isula pull does not support format name@digest

Signed-off-by: WangFengTu <wangfengtu@huawei.com>
---
 src/cmd/isula/images/pull.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cmd/isula/images/pull.c b/src/cmd/isula/images/pull.c
index 3ba7a715..da9cae52 100644
--- a/src/cmd/isula/images/pull.c
+++ b/src/cmd/isula/images/pull.c
@@ -25,7 +25,7 @@
 #include "connect.h"
 
 const char g_cmd_pull_desc[] = "Pull an image or a repository from a registry";
-const char g_cmd_pull_usage[] = "pull [OPTIONS] NAME[:TAG|@DIGEST]";
+const char g_cmd_pull_usage[] = "pull [OPTIONS] NAME[:TAG]";
 
 struct client_arguments g_cmd_pull_args = {};
 
-- 
2.25.1

