From 9663e62598570d16c0e8a70be4341ff72663b8df Mon Sep 17 00:00:00 2001
From: zhongtao <zhongtao17@huawei.com>
Date: Mon, 6 Mar 2023 09:51:17 +0800
Subject: [PATCH 34/35] change sleep() to usleep() to avoid lossing of accuracy

Signed-off-by: zhongtao <zhongtao17@huawei.com>
---
 src/utils/cutils/utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/cutils/utils.c b/src/utils/cutils/utils.c
index 2c4c01e4..983d81d8 100644
--- a/src/utils/cutils/utils.c
+++ b/src/utils/cutils/utils.c
@@ -345,7 +345,7 @@ int util_waitpid_with_timeout(pid_t pid, const int64_t timeout, handle_timeout_c
             return -1;
         }
         // sleep some time instead to avoid cpu full running and then retry.
-        sleep(0.1);
+        usleep(100);
     }
     return 0;
 }
-- 
2.25.1

