From 1bf460aa845611fbf20136eaa147b6c75f9419f1 Mon Sep 17 00:00:00 2001
From: Mig Yang <418109103@qq.com>
Date: Wed, 16 Nov 2022 17:32:18 +0800
Subject: [PATCH 44/45] fix can not install isulad rpm because of spec

---
 iSulad.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/iSulad.spec b/iSulad.spec
index e9826a29..97c6bcb3 100644
--- a/iSulad.spec
+++ b/iSulad.spec
@@ -18,6 +18,7 @@ ExclusiveArch:  x86_64 aarch64
 Provides:       libhttpclient.so()(64bit)
 Provides:       libisula.so()(64bit)
 Provides:       libisulad_img.so()(64bit)
+Provides:       libisulad_tools.so()(64bit)
 %endif
 
 %if 0%{?is_systemd}
@@ -79,6 +80,7 @@ cd build
 install -d $RPM_BUILD_ROOT/%{_libdir}
 install -m 0644 ./src/libisula.so             %{buildroot}/%{_libdir}/libisula.so
 install -m 0644 ./src/utils/http/libhttpclient.so  %{buildroot}/%{_libdir}/libhttpclient.so
+install -m 0644 ./src/libisulad_tools.so  %{buildroot}/%{_libdir}/libisulad_tools.so
 install -m 0644 ./src/daemon/modules/image/libisulad_img.so   %{buildroot}/%{_libdir}/libisulad_img.so
 chmod +x %{buildroot}/%{_libdir}/libisula.so
 chmod +x %{buildroot}/%{_libdir}/libhttpclient.so
-- 
2.25.1

