From 998835f4bca41a91b938a97d4a25e7389e24b19a Mon Sep 17 00:00:00 2001
From: WangFengTu <wangfengtu@huawei.com>
Date: Thu, 15 Apr 2021 09:45:06 +0800
Subject: [PATCH 073/104] update libarchive requirement to v3.4

Signed-off-by: WangFengTu <wangfengtu@huawei.com>
---
 cmake/checker.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/checker.cmake b/cmake/checker.cmake
index 5ba4c63d..d4337a1b 100644
--- a/cmake/checker.cmake
+++ b/cmake/checker.cmake
@@ -57,7 +57,7 @@ find_library(LIBYAJL_LIBRARY yajl
 _CHECK(LIBYAJL_LIBRARY "LIBYAJL_LIBRARY-NOTFOUND" "libyajl.so")
 
 # check libarchive
-pkg_check_modules(PC_LIBARCHIVE REQUIRED "libarchive>=3.2")
+pkg_check_modules(PC_LIBARCHIVE REQUIRED "libarchive>=3.4")
 find_path(LIBARCHIVE_INCLUDE_DIR archive.h
 	HINTS ${PC_LIBARCHIVE_INCLUDEDIR} ${PC_LIBARCHIVE_INCLUDE_DIRS})
 _CHECK(LIBARCHIVE_INCLUDE_DIR "LIBARCHIVE_INCLUDE_DIR-NOTFOUND" "archive.h")
-- 
2.25.1

