From 4e9f8ec1f3229e6992ab2750fac61a062bae64ed Mon Sep 17 00:00:00 2001
From: gaohuatao <gaohuatao@huawei.com>
Date: Tue, 27 Apr 2021 16:31:08 +0800
Subject: [PATCH 088/104] CI devicemapper add filter

Signed-off-by: gaohuatao <gaohuatao@huawei.com>
---
 CI/test_cases/helpers.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/CI/test_cases/helpers.sh b/CI/test_cases/helpers.sh
index c8e7bb8a..d40e61d9 100755
--- a/CI/test_cases/helpers.sh
+++ b/CI/test_cases/helpers.sh
@@ -196,9 +196,10 @@ function do_install_thinpool()
 {
     local ret=0
 
-    systemctl restart lvm2-lvmetad.service
-    systemctl restart systemd-udevd.service
-    udevadm control --reload-rules && udevadm trigger
+    cat /etc/isulad/daemon.json | grep driver | grep devicemapper
+    if [[ $? -ne 0 ]]; then
+        return ${ret}
+    fi
 
     dev_disk=`pvs | grep isulad | awk '{print$1}'`
     rm -rf /var/lib/isulad/*
-- 
2.25.1

