From 8003284bbf9d679e2d3f52cb55cdd4ee70d22977 Mon Sep 17 00:00:00 2001
From: lifeng68 <lifeng68@huawei.com>
Date: Thu, 29 Apr 2021 12:38:44 +0800
Subject: [PATCH 091/104] CI: run the containers one by one

Signed-off-by: lifeng68 <lifeng68@huawei.com>
---
 CI/build.sh | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/CI/build.sh b/CI/build.sh
index cf7691d9..5ebe8ea3 100755
--- a/CI/build.sh
+++ b/CI/build.sh
@@ -415,18 +415,17 @@ rm -rf ${cptemp}
 # wait for copy files become effective
 sleep 3
 
+docker exec ${copycontainer} tail -f --retry /tmp/runflag/${CONTAINER_NAME}.scripts.log 2>/dev/null &
+tailpid=$!
+
 for container in ${containers[@]}
 do
     {
         exec_script ${container} ${testcase_script}
-    }&
-    pids="$! $pids"
+    }
 done
 
-docker exec ${copycontainer} tail -f --retry /tmp/runflag/${CONTAINER_NAME}.scripts.log 2>/dev/null &
-tailpid=$!
 trap "kill -9 $tailpid; exit 0" 15 2
-wait $pids
 
 pid_dev="NULL"
 if [[ "x$disk" != "xNULL" ]] && [[ "x${enable_gcov}" == "xON" ]]; then
-- 
2.25.1

